packages/web/src/common/hooks/useFeatureFlag.ts

```diff
@@ -1,44 +1,78 @@
-import { useMemo } from 'react'
-
-import { useSelector } from 'react-redux'
+import { useCallback, useEffect, useMemo, useState } from 'react'
 
 import {
   FeatureFlags,
-  FeatureFlagCohortType,
-  RemoteConfigInstance,
-  flagCohortType
+  RemoteConfigInstance
 } from 'common/services/remote-config'
-import { getAccountUser } from 'common/store/account/selectors'
-import { isRemoteConfigLoaded } from 'common/store/remote-config/selectors'
-import { StateWithRemoteConfig } from 'common/store/remote-config/slice'
 
 export const FEATURE_FLAG_OVERRIDE_KEY = 'FeatureFlagOverride'
 
 export type OverrideSetting = 'enabled' | 'disabled' | null
 
+/**
+ * Helper for when to recompute flag state, used by both FeatureFlags
+ * and RemoteConfig. Recomputes when:
+ * - User logs in (account is seen in store)
+ * - Config loads
+ * - User ID is set on Optimizely (seen by event emission)
+ **/
+const useRecomputeBool = (
+  useAccountProvider: () => boolean,
+  configLoaded: boolean,
+  remoteConfigInstance: RemoteConfigInstance
+) => {
+  const [shouldRecompute, setShouldRecompute] = useState(false)
```

Comment:
useToggle might be nice here, from react-use, but actually i think the way you have might be better